tools/config-gen/params.yaml

```diff
@@ -15,16 +31,308 @@
+  config-path: "metadata-cache.stat-cache-max-size-mb"
+  type: "int"
+  usage: >-
+    The maximum size of stat-cache in MiBs. It can also be set to -1 for
+    no-size-limit, 0 for no cache. Values below -1 are not supported.
+  default: "math.MinInt64"
+
+- flag-name: "type-cache-max-size-mb"
+  config-path: "metadata-cache.type-cache-max-size-mb"
+  type: "int"
+  usage: "Max size of type-cache maps which are maintained at a per-directory level."
+  default: "4"
+
+- flag-name: "metadata-cache-ttl"
+  config-path: "metadata-cache.ttl-secs"
+  type: "int"
+  usage: >-
+    The ttl value in seconds to be used for expiring items in metadata-cache. It
+    can be set to -1 for no-ttl, 0 for no cache and > 0 for ttl-controlled
+    metadata-cache. Any value set below -1 will throw an error."
+  default: "math.MinInt64"
+
+- flag-name: "stat-cache-capacity"
+  config-path: "metadata-cache.deprecated-stat-cache-capacity"
+  type: "int"
+  usage: >-
+    How many entries can the stat-cache hold (impacts memory consumption). This
+    flag has been deprecated (starting v2.0) and in favor of
+    stat-cache-max-size-mb. For now, the value of stat-cache-capacity will be
+    translated to the next higher corresponding value of stat-cache-max-size-mb
+    (assuming stat-cache entry-size ~= 1640 bytes, including 1400 for positive
+    entry and 240 for corresponding negative entry), if stat-cache-max-size-mb
+    is not set."
+  deprecated: true
+  deprecation-warning: >-
+    This flag has been deprecated (starting v2.0) in favor of
+    stat-cache-max-size-mb.
+  default: "20460"
+
+- flag-name: "stat-cache-ttl"
+  config-path: "metadata-cache.deprecated-stat-cache-ttl"
+  type: "duration"
+  usage: >-
+    How long to cache StatObject results and inode attributes. This flag
+    has been deprecated (starting v2.0) in favor of metadata-cache-ttl-secs. For
+    now, the minimum of stat-cache-ttl and type-cache-ttl values, rounded up to
+    the next higher multiple of a second is used as ttl for both stat-cache and
+    type-cache, when metadata-cache-ttl-secs is not set.
+  default: "60s"
+  deprecated: true
+  deprecation-warning: >-
+    This flag has been deprecated (starting v2.0) in favor of
+    metadata-cache-ttl-secs.
+
+- flag-name: "type-cache-ttl"
+  config-path: "metadata-cache.deprecated-type-cache-ttl"
+  type: "duration"
+  usage: >-
+    Usage: How long to cache StatObject results and inode attributes. This flag
+    has been deprecated (starting v2.0) in favor of metadata-cache-ttl-secs. For
+    now, the minimum of stat-cache-ttl and type-cache-ttl values, rounded up to
+    the next higher multiple of a second is used as ttl for both stat-cache and
+    type-cache, when metadata-cache-ttl-secs is not set.
+  default: "60s"
+  deprecated: true
+  deprecation-warning: >-
+    This flag has been deprecated (starting v2.0) in favor of
+    metadata-cache-ttl-secs.
+
+- flag-name: "kernel-list-cache-ttl-secs"
```

Comment:
Should the flag be `kernel-list-cache-ttl` with type duration?

Reply:
It's currently an int flag: https://github.com/GoogleCloudPlatform/gcsfuse/blob/9cfb931f63dcc5559cee29aa2a31fb295ac620af/internal/config/mount_config.go#L89C2-L89C27

Reply:
my bad.. I am guessing we kept it this way to support `-1`. Resolving.